src/components/Buttons/Buttons.jsx

```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+
+export const Buttons = ({
+  buttonText,
+  url,
+  icon,
+  hoverIcon,
+  alt,
+  className,
+}) => {
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
+
+  return (
+    <a href={url} target="_blank" rel="noopener noreferrer">
+      <div
+        className={`buttons ${isHovered ? "hovered" : ""}`}
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handleMouseLeave}
+      >
```

Comment:
Nice. This is something new for me. I never use this kid of function for a hover. Good job